manifests/init.pp

```diff
@@ -1,21 +1,35 @@
 class wireguard (
-  Array[Hash] $tunnels = [],
-  Array[Hash] $simple_tunnels = [],
+  Hash[String, Struct[
+    {
+      ensure      => Optional[String],
+      private_key => String,
+      listen_port => Integer,
+      virtual_ip  => String,
+      peers       => Hash,
+    }
+  ]] $tunnels = {},
+  Hash[String, Struct[
+    {
+      ensure           => Optional[String],
+      private_key      => String,
+      listen_port      => Integer,
+      virtual_ip       => String,
+      peer_public_key  => String,
+      peer_allowed_ips => Optional[String],
+      peer_endpoint    => Optional[String],
+    }
+  ]] $simple_tunnels = [],
 ) {
   include wireguard::packages
 
-  $tunnels.each |$tunnel| {
-    $tunnel.each |$name, $params| {
-      wireguard::tunnel { $name:
-        * => $params,
-      }
+  $tunnels.each |$name, $params| {
+    wireguard::tunnel { $name:
+      * => $params,
```

Comment:
I'm not exactly sure why I wanted to do it in first place. Will have to check more thoroughly :-)